### PR TITLE
fix cross preference issue

### PR DESCRIPTION
**Pull request by bianliu1013**

### CODE REVIEW

File: docfx_yaml/extension.py [hunk 1]
```diff
@@ -399,6 +400,17 @@ def find_node_in_toc_tree(toc_yaml, to_add_node):
         )
 
 
+def missing_reference(app, env, node, contnode):
+    reftarget = ''
+    refdoc = ''
+    if 'refdomain' in node.attributes and node.attributes['refdomain'] == 'py':
+        reftarget = node['reftarget']
+        if reftarget.startswith('cntk.'):
```
Comment (superyyrrzz) [inline docfx_yaml/extension.py]: >cntk. [](start = 33, length = 5)

I think the tool should not be specific to cntk

File: docfx_yaml/writer.py [hunk 2]
```diff
@@ -878,7 +878,11 @@ def visit_reference(self, node):
             if 'http' in node.attributes['refuri']:
                 self.add_text('[{}]({})'.format(node.astext(), node.attributes['refuri']))
             else:
-                self.add_text('@{}'.format(node.attributes['refuri'].replace('#', '')))
+                # no need '.html#id' ending in yml files
+                pos = node.attributes['refuri'].find('.html')
+                if pos != -1:
+                     node.attributes['refuri'] = node.attributes['refuri'][0: pos]
```
Comment (superyyrrzz) [inline docfx_yaml/writer.py]: >node.attributes['refuri'][0: pos] [](start = 49, length = 33)

Is it more accurate to use 'id'?

File: docfx_yaml/writer.py [hunk 3]
```diff
@@ -878,7 +878,11 @@ def visit_reference(self, node):
             if 'http' in node.attributes['refuri']:
                 self.add_text('[{}]({})'.format(node.astext(), node.attributes['refuri']))
             else:
-                self.add_text('@{}'.format(node.attributes['refuri'].replace('#', '')))
+                # no need '.html#id' ending in yml files
```
Comment (superyyrrzz) [inline docfx_yaml/writer.py]: add a todo here...

File: docfx_yaml/extension.py (hunk 1 above)
Comment (bianliu1013) [reply docfx_yaml/extension.py]: removed

File: docfx_yaml/writer.py (hunk 2 above)
Comment (bianliu1013) [reply docfx_yaml/writer.py]: no need. 'refuri' return string somelike:  [target].html#segment. so we need to remove ".html#segment" as template don't need them.

File: docfx_yaml/writer.py (hunk 3 above)
Comment (bianliu1013) [reply docfx_yaml/writer.py]: added.

**Review by superyyrrzz — APPROVED** [review]

**Pull request merged**